site/guides/variant-constructor-pattern.md

```diff
@@ -0,0 +1,182 @@
+# Variant construction pattern
+
+If you look at the config workflows, you will notice that creating variant
+of a package is a very common and frequent operation, so reducing the steps
+required to create a variant can have significant benefits for the
+package consumers. In this guide, we will look at some techniques
+that a package author can use to enable automatic variant construction of a package.
+
+## Types of packages
+
+kpt packages comes in two flavors:  `abstract package` and
+`deployable package instance`. An `abstract` package is a reususable package that
```

Comment:
Here we use the phrase `deployable package instance`, and then later we just refer to it as `deployable instance`.  Shall we stick with one phrasing to make it abundantly clear/consistent?